replacing index.html hard-code with an rST page

choldgraf opened the pull request:
Currently our landing page is hard-coded with HTML (https://github.com/jupyter/jupyter/blob/master/docs/source/_templates/index.html). This PR converts that page into an `index.rst` page instead, so that it is more readable and maintainable over time (it moves `contents.rst` to `index.rst` so we have an index page now). It also fixes some of the CSS issues that we currently have on the front page.

Here's a preview: https://80-36895421-gh.circle-artifacts.com/0/html/index.html

GeorgianaElena commented on docs/source/index.rst:
```diff
@@ -0,0 +1,201 @@
+- :ref:`search`
+
+
+**Resources:**
```
```suggestion
Resources
========
```
What do you think about having `resources` as a subtitle?

choldgraf commented:
ok, I rebased to add in @GeorgianaElena's link fixes and also moved around the resources section. @GeorgianaElena do you think that the new and old pages look similar enough?

GeorgianaElena commented:
Yes! I really like how the table for the sub-project docs looks.
Regarding this issue https://github.com/jupyter/jupyter/issues/509, is there a way to check if the translations still work with this new setup?

choldgraf commented:
I'm not sure, I think @parente set up all of the translation infrastructure, and I'm not super familiar with it.

Although, I *think* that the landing page is the same between the languages. E.g., compare

https://jupyter.readthedocs.io/es/latest/

and

https://jupyter.readthedocs.io/en/latest/

do we see any differences?

GeorgianaElena commented:
Yes, I think you're right, other than *Contents* nothing else gets translated.
But because of all the `{%trans%}` appearances in `index.html`, I think the intention was to translate all the words in the page.

I think it should have been a `*.po` file for the landing page in the corresponding language ([here](https://github.com/jupyter/jupyter/tree/master/docs/source/locale/es/LC_MESSAGES)  for example).
And I think the reason why only *Contents* gets translated is because there is a mapping for the word in the other `*.po` files (https://github.com/jupyter/jupyter/search?q=contenidos&unscoped_q=contenidos)

P.S. I've never seen a *.po file before, so I'm just guessing

choldgraf commented:
OK cool - hmmm, so maybe we should give it a day or two to see if @parente objects to merging and if not we can go for it? I am happy to try figuring out how to make `index.rst` translatable if this raises issues...just am not familiar with Transifex myself

parente commented:
I just got around to handling my GitHub notifications for the week and commented in the related issue. 

Go ahead and merge this PR as is. The automation set up in #430 should create the necessary .po template files and make them available for translation in Transifex. If we have `index.html` translations for languages that are not yet configured in Transifex, I can help set those up.

I do have a local branch with a start on documentation about how the translation CI/CD works. I struggled to integrate it in to the existing structure of the current website. Figuring out where it should live is something I can help with in #511.

choldgraf commented:
ok great - I just rebased, and am gonna merge! I'll try to spend some time making other iterative improvements this and next week. @parente let me know how I can help on the translation side.

parente commented:
The automation worked:

1. Travis [created a commit](https://github.com/jupyter/jupyter/commit/1330bc409842d8b8a7bbb3a1c63259c34a543be0) for a new [translation template for `index.rst`](https://github.com/jupyter/jupyter/blob/master/docs/source/locale/en/LC_MESSAGES/index.po)
2. The template became available for translation in Transifex by teams and locales configured there: https://www.transifex.com/project-jupyter/jupyter-meta-documentation/docs-source-locale-en-lc-messages-index-po--master/

@choldgraf It looks like I added you as an admin to the Jupyter org on transifex a while ago along with Carol. If you want to poke around before I draft the doc, https://www.transifex.com/project-jupyter/ is the place to visit.

choldgraf commented:
cool! what was the email you used to add me? I can try digging around for an email about it

parente commented:
I think it's just tied to your GitHub account if you login using GitHub SSO.

choldgraf commented:
got it working!

parente commented:
FWIW, it's possible this PR broke the links under the `Start Here` heading on the first page.  The links in the top nav bar still function, but it looks like the ones in the main part of the page all lead to the "this page does not exist yet" error page from RTD.

choldgraf commented:
good catch - I just fixed this in #516
